File: frontend/src/api/fitness.ts
import { apiFetch } from "./client"

export interface FitnessPoint {
  date: string
  ctl: number | null
  atl: number | null
  tsb: number | null
  ramp_rate: number | null
}

export function fetchFitnessSeries(athleteId: number, days = 90): Promise<FitnessPoint[]> {
  return apiFetch<FitnessPoint[]>(`/api/v1/athlete/${athleteId}/fitness?days=${days}`)
}

File: sidekick/src/models/fitness.py
from pydantic import BaseModel


class FitnessPoint(BaseModel):
    """One day of fitness/fatigue/form, sourced from Intervals.icu's wellness data.

    Not persisted — a plain response shape, not a Mongo document. TSB is derived
    locally (ctl - atl); it isn't an Intervals.icu-specific field.
    """

    date: str  # YYYY-MM-DD
    ctl: float | None = None
    atl: float | None = None
    tsb: float | None = None
    ramp_rate: float | None = None
